legacy/reqcounter/reqcounter.go

```diff
@@ -16,30 +16,73 @@ limitations under the License.
 
 package reqcounter
 
-import "sync"
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
 
 // RequestCounter records the number of HTTP requests to GCR.
 type RequestCounter struct {
-	mutex    sync.Mutex
-	requests uint64
+	mutex    sync.Mutex // Lock to prevent race-conditions with concurrent processes.
+	requests uint64     // Number of HTTP requests since recording started.
+	quota    uint64     // Maximum requests allowed before GCR throttling occurs.
```

Comment:
This field shouldn't be here, because it's already a global, read-only constant in the module.

Reply:
I included this in the RequestCounter for easier testing that relies solely on the object, not global constants. But I can get around this.